python_dwd/download/download.py

```diff
@@ -45,24 +47,25 @@ def _download_dwd_data(remote_file: Union[str, Path]) -> BytesIO:
         stores data on local file system
 
     """
+    assert isinstance(remote_file, str)
 
     file_server = create_remote_file_name(remote_file)
 
     try:
-        with urllib.request.urlopen(file_server) as url_request:
+        with urlopen(file_server) as url_request:
             zip_file = BytesIO(url_request.read())
-    except urllib.error.URLError as e:
-        raise urllib.error.URLError(f"Error: the stationdata {file_server} couldn't be reached.\n"
-                                    f"{str(e)}")
+    except URLError:
+        raise URLError(f"Error: the stationdata {file_server} couldn't be reached.")
+    except IOError:
+        raise IOError(f"Error: the stationdata {file_server} couldn't be read into a bytes object.")
 
     try:
         with zipfile.ZipFile(zip_file) as zip_file_opened:
```

Comment:
Handling could be much easier if you extratc this into a a function. 
define function with an _ at the beginning. 
E.g.: 
```
def _zip_file_extraction(zip_file):
        with zipfile.ZipFile(zip_file) as zip_file_opened:
                produkt_file = [file_in_zip
￼                            for file_in_zip in zip_file_opened.namelist()
￼                            if find_all_matchstrings_in_string(file_in_zip, STATIONDATA_MATCHSTRINGS)][0]
￼        return BytesIO(zip_file_opened.open(produkt_file).read())

```

Reply:
Advantage: You can test this seperately